Not building on Xubuntu 15.10

Exactly what it says on the tin. I definitely have the development libs installed. 15.10 has udev 225. Here's my make output:

```
WC130-TiZ:MoltenGamepad$ make
g++  -std=c++14 -ludev -lpthread -I source source/*.cpp source/*/*.cpp source/*/*/*.cpp -o moltengamepad
source/uinput.cpp: In member function ‘int uinput::make_gamepad(const uinput_ids&, bool, bool)’:
source/uinput.cpp:55:71: warning: format not a string literal and no format arguments [-Wformat-security]
   snprintf(uidev.name, UINPUT_MAX_NAME_SIZE, ids.device_string.c_str());
                                                                       ^
source/uinput.cpp:55:71: warning: format not a string literal and no format arguments [-Wformat-security]
source/uinput.cpp: In member function ‘int uinput::make_keyboard(const uinput_ids&)’:
source/uinput.cpp:130:71: warning: format not a string literal and no format arguments [-Wformat-security]
   snprintf(uidev.name, UINPUT_MAX_NAME_SIZE, ids.device_string.c_str());
                                                                       ^
source/uinput.cpp:130:71: warning: format not a string literal and no format arguments [-Wformat-security]
/tmp/ccDTOSq9.o: In function `std::thread::thread<void (exclusive_chord::*)(), exclusive_chord*>(void (exclusive_chord::*&&)(), exclusive_chord*&&)':
event_change.cpp:(.text._ZNSt6threadC2IM15exclusive_chordFvvEJPS1_EEEOT_DpOT0_[_ZNSt6threadC5IM15exclusive_chordFvvEJPS1_EEEOT_DpOT0_]+0x93): undefined reference to `pthread_create'
/tmp/cc00bN2L.o: In function `std::thread::thread<int (&)(moltengamepad*), moltengamepad*>(int (&)(moltengamepad*), moltengamepad*&&)':
moltengamepad.cpp:(.text._ZNSt6threadC2IRFiP13moltengamepadEJS2_EEEOT_DpOT0_[_ZNSt6threadC5IRFiP13moltengamepadEJS2_EEEOT_DpOT0_]+0x93): undefined reference to `pthread_create'
/tmp/ccQzM8mx.o: In function `udev_handler::pass_along_device(udev_device*)':
udev.cpp:(.text+0x101): undefined reference to `udev_device_get_syspath'
/tmp/ccQzM8mx.o: In function `udev_handler::udev_handler()':
udev.cpp:(.text+0x277): undefined reference to `udev_new'
/tmp/ccQzM8mx.o: In function `udev_handler::~udev_handler()':
udev.cpp:(.text+0x33d): undefined reference to `udev_monitor_unref'
udev.cpp:(.text+0x35a): undefined reference to `udev_unref'
/tmp/ccQzM8mx.o: In function `udev_handler::start_monitor()':
udev.cpp:(.text+0x3cd): undefined reference to `udev_monitor_new_from_netlink'
udev.cpp:(.text+0x3f2): undefined reference to `udev_monitor_filter_add_match_subsystem_devtype'
udev.cpp:(.text+0x40c): undefined reference to `udev_monitor_filter_add_match_subsystem_devtype'
udev.cpp:(.text+0x41c): undefined reference to `udev_monitor_enable_receiving'
/tmp/ccQzM8mx.o: In function `udev_handler::enumerate()':
udev.cpp:(.text+0x4c2): undefined reference to `udev_enumerate_new'
udev.cpp:(.text+0x4d7): undefined reference to `udev_enumerate_add_match_subsystem'
udev.cpp:(.text+0x4e8): undefined reference to `udev_enumerate_add_match_subsystem'
udev.cpp:(.text+0x4f4): undefined reference to `udev_enumerate_scan_devices'
udev.cpp:(.text+0x500): undefined reference to `udev_enumerate_get_list_entry'
udev.cpp:(.text+0x51f): undefined reference to `udev_list_entry_get_name'
udev.cpp:(.text+0x53a): undefined reference to `udev_device_new_from_syspath'
udev.cpp:(.text+0x55d): undefined reference to `udev_device_unref'
udev.cpp:(.text+0x569): undefined reference to `udev_list_entry_get_next'
udev.cpp:(.text+0x57b): undefined reference to `udev_enumerate_unref'
/tmp/ccQzM8mx.o: In function `udev_handler::read_monitor()':
udev.cpp:(.text+0x5f5): undefined reference to `udev_monitor_get_fd'
udev.cpp:(.text+0x6ac): undefined reference to `udev_monitor_receive_device'
udev.cpp:(.text+0x6d6): undefined reference to `udev_device_unref'
/tmp/ccQzM8mx.o: In function `std::thread::thread<int (udev_handler::*)(), udev_handler*>(int (udev_handler::*&&)(), udev_handler*&&)':
udev.cpp:(.text._ZNSt6threadC2IM12udev_handlerFivEJPS1_EEEOT_DpOT0_[_ZNSt6threadC5IM12udev_handlerFivEJPS1_EEEOT_DpOT0_]+0x93): undefined reference to `pthread_create'
/tmp/ccgS5ifT.o: In function `std::thread::thread<void (input_source::*)(), input_source*>(void (input_source::*&&)(), input_source*&&)':
input_source.cpp:(.text._ZNSt6threadC2IM12input_sourceFvvEJPS1_EEEOT_DpOT0_[_ZNSt6threadC5IM12input_sourceFvvEJPS1_EEEOT_DpOT0_]+0x93): undefined reference to `pthread_create'
/tmp/cchzvL6l.o: In function `generic_device::~generic_device()':
generic_input.cpp:(.text+0x461): undefined reference to `udev_device_unref'
/tmp/ccAogSsL.o: In function `generic_manager::accept_device(udev*, udev_device*)':
generic_manager.cpp:(.text+0xa84): undefined reference to `udev_device_get_syspath'
generic_manager.cpp:(.text+0xa94): undefined reference to `udev_device_get_subsystem'
generic_manager.cpp:(.text+0xaa4): undefined reference to `udev_device_get_action'
generic_manager.cpp:(.text+0xc33): undefined reference to `udev_device_get_sysname'
generic_manager.cpp:(.text+0xc69): undefined reference to `udev_device_get_parent'
generic_manager.cpp:(.text+0xc7e): undefined reference to `udev_device_get_sysattr_value'
/tmp/ccAogSsL.o: In function `generic_file::open_node(udev_device*)':
generic_manager.cpp:(.text._ZN12generic_file9open_nodeEP11udev_device[_ZN12generic_file9open_nodeEP11udev_device]+0x43): undefined reference to `udev_device_get_devnode'
generic_manager.cpp:(.text._ZN12generic_file9open_nodeEP11udev_device[_ZN12generic_file9open_nodeEP11udev_device]+0xdc): undefined reference to `udev_device_get_devnode'
generic_manager.cpp:(.text._ZN12generic_file9open_nodeEP11udev_device[_ZN12generic_file9open_nodeEP11udev_device]+0x17e): undefined reference to `udev_device_get_devnode'
generic_manager.cpp:(.text._ZN12generic_file9open_nodeEP11udev_device[_ZN12generic_file9open_nodeEP11udev_device]+0x23a): undefined reference to `udev_device_ref'
/tmp/ccAogSsL.o: In function `generic_file::close_node(udev_device*, bool)':
generic_manager.cpp:(.text._ZN12generic_file10close_nodeEP11udev_deviceb[_ZN12generic_file10close_nodeEP11udev_deviceb]+0x2d): undefined reference to `udev_device_get_devnode'
/tmp/ccAogSsL.o: In function `generic_file::close_node(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)':
generic_manager.cpp:(.text._ZN12generic_file10close_nodeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEb[_ZN12generic_file10close_nodeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEb]+0xc6): undefined reference to `udev_device_unref'
/tmp/ccAogSsL.o: In function `std::thread::thread<void (generic_file::*)(), generic_file*>(void (generic_file::*&&)(), generic_file*&&)':
generic_manager.cpp:(.text._ZNSt6threadC2IM12generic_fileFvvEJPS1_EEEOT_DpOT0_[_ZNSt6threadC5IM12generic_fileFvvEJPS1_EEEOT_DpOT0_]+0x93): undefined reference to `pthread_create'
/tmp/ccY00ulV.o: In function `wiimote::clear_node(dev_node*)':
wiimote.cpp:(.text+0x91d): undefined reference to `udev_device_unref'
/tmp/ccY00ulV.o: In function `wiimote::handle_event(udev_device*)':
wiimote.cpp:(.text+0x970): undefined reference to `udev_device_get_subsystem'
wiimote.cpp:(.text+0x980): undefined reference to `udev_device_get_action'
wiimote.cpp:(.text+0x9ac): undefined reference to `udev_device_get_parent'
wiimote.cpp:(.text+0x9b9): undefined reference to `udev_device_get_sysattr_value'
wiimote.cpp:(.text+0xa58): undefined reference to `udev_device_get_sysname'
wiimote.cpp:(.text+0xa8a): undefined reference to `udev_device_get_parent'
wiimote.cpp:(.text+0xa9f): undefined reference to `udev_device_get_sysattr_value'
/tmp/ccY00ulV.o: In function `wiimote::store_node(udev_device*, char const*)':
wiimote.cpp:(.text+0xc7a): undefined reference to `udev_device_ref'
wiimote.cpp:(.text+0xcf9): undefined reference to `udev_device_ref'
wiimote.cpp:(.text+0xdbb): undefined reference to `udev_device_ref'
wiimote.cpp:(.text+0xe7d): undefined reference to `udev_device_ref'
wiimote.cpp:(.text+0xeaa): undefined reference to `udev_device_ref'
/tmp/ccY00ulV.o:wiimote.cpp:(.text+0xf43): more undefined references to `udev_device_ref' follow
/tmp/ccY00ulV.o: In function `wiimote::open_node(dev_node*)':
wiimote.cpp:(.text+0x1543): undefined reference to `udev_device_get_devnode'
/tmp/ccY00ulV.o: In function `wiimote::grab_chmod_node(dev_node*, bool)':
wiimote.cpp:(.text+0x1663): undefined reference to `udev_device_get_devnode'
wiimote.cpp:(.text+0x168e): undefined reference to `udev_device_get_devnode'
/tmp/ccJqVgPA.o: In function `find_wii_dev_by_path(std::vector<wiimote*, std::allocator<wiimote*> >*, char const*)':
wiimote_manager.cpp:(.text+0x199): undefined reference to `udev_device_get_syspath'
/tmp/ccJqVgPA.o: In function `destroy_wii_dev_by_path(moltengamepad*, std::vector<wiimote*, std::allocator<wiimote*> >*, char const*)':
wiimote_manager.cpp:(.text+0x288): undefined reference to `udev_device_get_syspath'
/tmp/ccJqVgPA.o: In function `wiimote_manager::accept_device(udev*, udev_device*)':
wiimote_manager.cpp:(.text+0x4b1): undefined reference to `udev_device_get_syspath'
wiimote_manager.cpp:(.text+0x4c7): undefined reference to `udev_device_get_subsystem'
wiimote_manager.cpp:(.text+0x4dd): undefined reference to `udev_device_get_action'
wiimote_manager.cpp:(.text+0x524): undefined reference to `udev_device_get_syspath'
wiimote_manager.cpp:(.text+0x5f2): undefined reference to `udev_device_get_parent_with_subsystem_devtype'
wiimote_manager.cpp:(.text+0x608): undefined reference to `udev_device_get_subsystem'
wiimote_manager.cpp:(.text+0x64f): undefined reference to `udev_device_get_driver'
wiimote_manager.cpp:(.text+0x688): undefined reference to `udev_device_get_syspath'
wiimote_manager.cpp:(.text+0x706): undefined reference to `udev_device_ref'
collect2: error: ld returned 1 exit status
Makefile:3: recipe for target 'all' failed
make: *** [all] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,11 @@
 
 all:
-	g++  -std=c++14 -ludev -lpthread -I source source/*.cpp source/*/*.cpp source/*/*/*.cpp -o moltengamepad
+	g++ -std=c++14 -I source source/*.cpp source/*/*.cpp source/*/*/*.cpp -o moltengamepad -ludev -lpthread
 
 clean:
 	rm ./moltengamepad
 debug:
-	g++ -g -std=c++14 -ludev -lpthread -I source source/*.cpp source/*/*.cpp source/*/*/*.cpp -o moltengamepad
+	g++ -g -std=c++14 -I source source/*.cpp source/*/*.cpp source/*/*/*.cpp -o moltengamepad -ludev -lpthread
 
 steam:
-	g++  -DBUILD_STEAM_CONTROLLER_DRIVER -std=c++14 -lscraw -ludev -lpthread -I source source/*.cpp source/*/*.cpp source/*/*/*.cpp -o moltengamepad
+	g++ -DBUILD_STEAM_CONTROLLER_DRIVER -std=c++14 -lscraw -I source source/*.cpp source/*/*.cpp source/*/*/*.cpp -o moltengamepad -ludev -lpthread

